src/systems/filecoin_nodes/network/_index.md

```diff
@@ -17,4 +17,6 @@ Here is the list of libp2p protocols used by Filecoin.
 - Bootstrap List: Bootstrap is a list of nodes that a new node attempts to connect upon joining the network. The list of bootstrap nodes and their addresses are defined by the users.
 - Peer Exchange: spec [TODO](https://github.com/libp2p/specs/issues/222)
 - DNSDiscovery: Design and spec needed before implementing
-- HTTPDiscovery: Design and spec needed before implementing 
\ No newline at end of file
+- HTTPDiscovery: Design and spec needed before implementing 
+
+Filecoin network protocols should be differentiated from non-filecoin network protocols.  Additionally different filecoin networks should differentiate protocols from each other.  To achieve this filecoin network protocol IDs MUST adhere to the following format: `/fil/<libp2p-protocol>/<network-name>`.  For example: `fil/kad-dht/mainnet`.
```

Comment:
We should definitely not do this for all protocols, things like bitswap, graphsync, and the hello protocol should work between all peers.

Maybe this makes sense for other protocols like deal making, or payment channel negotiation, or retrieval. But I think a better solution would be to include the network identifier in the message itself instead (though i feel less strongly about those cases)

Reply:
Any opinions on gossipsub, and dht?

Reply:
Also it sounds like this objection has to do with interop between filecoin and non-filecoin nodes.  It would be great to hear a quick sketch of intended use cases for the network that motivate this in the particular protocols you have flagged.  I'm guessing this is IPFS interop related?

Reply:
I concur that we should not namespace protocols that are not specific to a filecoin network, including bitswap and gossipsub.

I think we *should* namespace:
- the DHT
- Hello
- The message and block pubsub channels and blocksync
- Deal exchange, retrieval and data transfer (probably)

@ZenGround0 could you please expand this to an explicit list of protocols (either here or where the protocols are or should be defined)

@whyrusleeping we seem to disagree about hello protocol. What's the use case where we'd want nodes from different networks (e.g. a testnet and mainnet) be able to exchange hello messages?

Reply:
Heard back indirectly regarding hello.  @whyrusleeping wants to keep network name in hello message, potentially for error handling.

Benefit still unclear clear but this is a minor difference and trying to optimize by shutting down hello connections at protocol id reading time looks premature.  Also we need to spec the hello protocol.